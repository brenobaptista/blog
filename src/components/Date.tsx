import { parseISO, format } from 'date-fns'

interface Props {
  date: string
}

const Date = ({ date }: Props): JSX.Element => {
  const parsedDate = parseISO(date)

  return <time dateTime={date}>{format(parsedDate, 'LLLL d, yyyy')}</time>
}

export default Date
