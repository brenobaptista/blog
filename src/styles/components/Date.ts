import styled from 'styled-components'

const Time = styled.time`
  color: ${props => props.theme.colors.date};
`

export default Time
