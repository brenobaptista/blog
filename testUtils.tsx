import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
