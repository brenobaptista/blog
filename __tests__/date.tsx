import { render } from '../testUtils'
import Date from '../src/components/Date'

test('matches snapshot', () => {
  const { asFragment } = render(<Date date='2020-01-01' />)

  expect(asFragment()).toMatchSnapshot()
})
