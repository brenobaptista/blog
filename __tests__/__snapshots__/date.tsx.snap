// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #707070;
}

<time
    class="c0"
    datetime="2020-01-01"
  >
    January 1, 2020
  </time>
</DocumentFragment>
`;
